lib/tasks/migrate_statement_of_case.rake

```diff
@@ -0,0 +1,45 @@
+def count_mismatch?(records, upload_target, text_target)
+  upload_target -= remove_missing_socs?
+  [
+    records.where(upload: true).count != upload_target,
+    records.where(typed: true).count != text_target,
+  ].any?(true)
+end
+
+def remove_missing_socs?
+  # sometimes, the statement of case is not created - a bug has been investigated and this ticket
+  # is part of the hoped for fix - but it means there are at least 6 applications that have had a
+  # file uploaded but have no linked ApplicationMeritsTask::StatementOfCase record in the database
+  attachment_laa_ids = Attachment.statement_of_case.pluck(:legal_aid_application_id).uniq
+  soc_laa_ids = ApplicationMeritsTask::StatementOfCase.where(upload: true).pluck(:legal_aid_application_id).uniq
+  missing_socs = attachment_laa_ids - soc_laa_ids
+  all_missing = missing_socs.map { |id| LegalAidApplication.find(id).statement_of_case.nil? }.all?(true)
+  all_missing ? missing_socs.count : 0
+end
+
+namespace :migrate do
+  desc "AP-5910: Migrate the statement of case type information"
+  task statements_of_case: :environment do
+    Rails.logger.info "== Before migration"
+    records = ApplicationMeritsTask::StatementOfCase.all
+    text_target = records.where.not(statement: "").count
+    upload_target = Attachment.statement_of_case.pluck(:legal_aid_application_id).uniq.count
+    Rails.logger.info "Affected applications: #{records.count}"
+    Rails.logger.info "Number with text statement: #{text_target}"
+    Rails.logger.info "Number with files uploaded: #{upload_target}"
+
+    ActiveRecord::Base.transaction do
+      records.find_each do |statement|
+        typed = statement.statement.present?
+        upload = statement.legal_aid_application.statement_of_case_uploaded?
+        statement.update!(typed:, upload:)
```

Comment:
I just noted/remember somthing reviewing Adams rake task - it's probabkly best to not "touch" the statement of case to avoid delaying purging of the application. I don't think it will matter because it is not updating the application _but_ does activerecord cascade changes from associated records to the parent (the legal_aid application) 🤷 .



```
  statement.update!(typed:, upload:, touch: false) # this prevents the updated_at date being changed and delaying purging of stale records
```

I don't _think_ it will unless the belongs_to is defined as `touch: true`, which it is not.

Reply:
`touch: false` doesn't work on update, only `save`

Changed to `update_columns` which doesn't trigger callbacks or timestamps

Reply:
damn, wasn't sure if it worked on update or not - now i know.

Not normally keen on update_columns but can't see any related callbacks and validations are relevant here and assume this all worked against prod dump, in which case 👍